asv_bench/benchmarks/algorithms.py

```diff
@@ -12,113 +12,151 @@
     except:
         pass
 
-class Algorithms(object):
+
+class Factorize(object):
+
+    goal_time = 0.2
+
+    def setup(self):
+        N = 10**5
+        np.random.seed(1234)
+        self.int_idx = pd.Int64Index(np.arange(N).repeat(5))
+        self.float_idx = pd.Float64Index(np.random.randn(N).repeat(5))
+        self.string_idx = tm.makeStringIndex(N)
+
+    def time_factorize_int(self):
+        self.int_idx.factorize()
+
+    def time_factorize_float(self):
+        self.float_idx.factorize()
+
+    def time_factorize_string(self):
+        self.string_idx.factorize()
+
+
+class Duplicated(object):
+
     goal_time = 0.2
 
     def setup(self):
-        N = 100000
+        N = 10**5
         np.random.seed(1234)
+        self.int_idx = pd.Int64Index(np.arange(N).repeat(5))
+        self.float_idx = pd.Float64Index(np.random.randn(N).repeat(5))
```

Comment:
Breaking this into two classes it duplicating the setup, and the 'Factorize' and 'Duplicated' name is also already in the `time_..` method names. So not sure it is necessarily a net improvement

(for this specific case I actually also think we could add a `time_duplicated_string`)